src/Client/Connector.hpp

```diff
@@ -283,6 +278,10 @@ Connector<BUFFER, NetProvider>::waitAll(Connection<BUFFER, NetProvider> &conn,
 		}
 		if (finish)
 			return 0;
+		if (m_NetProvider.wait(timer.timeLeft()) != 0) {
+			conn.setError(std::string("Failed to poll: ") + strerror(errno), errno);
+			return -1;
+		}
```

Comment:
It looks weird to me that there is no check of readiness after the timeout is expired - it would be nice to return TimeOut error only if it's really the case.

Also, since we don't decode responses after the `wait` here, we will have to decode them on the next `Connector::wait*(...)` call, and it will reduce actual wait time (we will spend some time from timeout to decode previously received responses).